### DISCUSSION TITLE
Infinity Polling Should Handle Exceptions

### DISCUSSION THREAD

**Discussion by All-The-Foxes** [eternnoir/pyTelegramBotAPI#2378] [Ideas]
It is wild to me that any error, including 429 and 401, is ignored during polling. Polling needs to break for 401 errors (unauthorized) and listen to 429 errors (too many requests). There shouldn't be any "unhandled exceptions" in the logs, let alone like 20 of them at once any time there is a bad gateway. Every exception, no matter what it is, should always be handled.

**Comment by coder2020official** [comment]
You mean for async?

**Reply by All-The-Foxes** [reply]
Yes.

**Comment by coder2020official** [comment]
By "not handled" in async, we refer to the fact that your exception handler did not handle this exception. This will not crush the code if you have non_stop set to True.

**Reply by All-The-Foxes** [reply]
An I supposed to use my own handler for the async version?

**Reply by coder2020official** [reply]
If you wish so.
Honestly, it is just a wording difference. While sync says "exception occured" or smth, async says it isn't handled as an addition I think

**Reply by All-The-Foxes** [reply]
Yes, async says unhandled exception, but the exceptions are also not handled. 429 errors are ignored, polling retries despite getting 401 errors, etc. They should be handled.

**Reply by coder2020official** [reply]
What do you mean by 'exceptions are also not handled'

**Reply by All-The-Foxes** [reply]
I don't know how else to say it. Exceptions are logged as tracebacks and not handled.

**Comment by All-The-Foxes** [comment]
To add to this, a full traceback for things like request timeouts just makes a mess in the logs. We don't need a traceback for Telegram server issues like bad gateways and such.
